implementation/parallel-to-sns-scatter-gather/scatter_gather/stepfunction/infrastructure.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python3
+# -*- coding:utf-8 -*-
+###
+# File: infrastructure.py
+# Created Date: Tuesday, January 24th 2023, 11:32:48 am
+# Author: Agostino Di Figlia
+# -----
+# Copyright (c) 2023 Amazon Web Services
+# 
+# 2022 Amazon Web Services, Inc. or its affiliates. All Rights Reserved.
+# This AWS Content is provided subject to the terms of the AWS Customer Agreement available at  
+# http://aws.amazon.com/agreement or other written agreement between Customer and either
+# Amazon Web Services, Inc. or Amazon Web Services EMEA SARL or both.
+# Note:
+# THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+###
+from constructs import Construct
+from aws_cdk import (
+    aws_stepfunctions as sfn,
+    aws_stepfunctions_tasks as sfn_tasks,
+    aws_iam as iam,
+    aws_logs as log,
+    Duration
+)
+import json
+import os
+import pathlib
+
+from scatter_gather.lambda_.infrastructure import LambdaStates
+
+
+class SFNWorkflow(Construct):
```

Comment:
I am probably not looking the right place, but wouldn't we need  two SFN workflows, one original and one refactored?

Reply:
refactored solution does not use orchestration but choreography (as discussed)